src/js/RPE.ts

```diff
@@ -0,0 +1,159 @@
+
```

Comment:
If we're going to do all of the work to make this a well-written, maintainable application, I would like to request that we also introduce a linter. Fine with eslint and standard airbnb if that works for you? Happy to open a PR against your PR with those changes. I'm a huge stickler for style. 🤐

Reply:
TypeScript does a lot of great "linting" already regarding function usage, types, some basic rules (and is enforced and build time- i.e. just won't compile). For everything else we can introduce TSlint which is comminly used with TS (https://palantir.github.io/tslint/rules/)